src/net.cpp

```diff
@@ -1345,7 +1373,16 @@ void CConnman::SocketHandlerConnected(const std::vector<CNode*>& nodes,
         if (sendSet) {
             // Send data
             size_t bytes_sent = WITH_LOCK(pnode->cs_vSend, return SocketSendData(*pnode));
-            if (bytes_sent) RecordBytesSent(bytes_sent);
+            if (bytes_sent) {
+                RecordBytesSent(bytes_sent);
+                // RecordMsgStatsSent(msg.m_type, bytes_sent, pnode->m_conn_type, pnode->ConnectedThroughNetwork());
+
+                // TODO how to get the message type here? This code will be hit if CConnman::PushMessage has messages on the queue
+                // that get sent now by SocketSendData.
+                // One option is to just record as sent once CConnman::PushMessage() puts them on the CNode.vSendMsg queue.
+                // Somewhere around this line in CConnman::PushMessage():
+                //        pnode->mapSendBytesPerMsgType[msg.m_type] += nTotalSize;
+            }
```

Comment:
Yes, I think there is no need to do anything here and we should only record it as `mapSendBytesPerMsgType[]` even though it would mean that we record some bytes as sent before them being actually sent (and they may never be sent). That seems acceptable given the complexity otherwise, if we want to record after the send. And this is how `mapSendBytesPerMsgType[]` works as well.